packages/editor/src/components/blocks/datatable/DataTable.tsx

```diff
@@ -0,0 +1,92 @@
+import type { DataTable, Prettify, Variable } from '@axonivy/form-editor-protocol';
+import type { ComponentConfig, UiComponentProps } from '../../../types/config';
+import './DataTable.css';
+import { baseComponentFields, defaultBaseComponent } from '../base';
+import IconSvg from './DataTable.svg?react';
+import { ComponentBlock } from '../../../editor/canvas/ComponentBlock';
+import { useAppContext } from '../../../context/AppContext';
+import { useEffect, useState } from 'react';
+import { Button, Message, type BrowserNode } from '@axonivy/ui-components';
+import { useMeta } from '../../../context/useMeta';
+import { findAttributesOfType } from '../../../data/variable-tree-data';
+import { componentByName } from '../../components';
+import { createInitiTableColumns } from '../../../data/data';
+import { IvyIcons } from '@axonivy/ui-icons';
+
+type DataTableProps = Prettify<DataTable>;
+
+export const defaultDataTableProps: DataTable = {
+  components: [],
+  label: 'Label',
+  value: '',
+  ...defaultBaseComponent
+} as const;
+
+export const DataTableComponent: ComponentConfig<DataTableProps> = {
+  name: 'DataTable',
+  category: 'Elements',
+  subcategory: 'Input',
+  icon: <IconSvg />,
+  description: 'A datatable component',
+  defaultProps: defaultDataTableProps,
+  render: props => <UiBlock {...props} />,
+  create: ({ label, value, ...defaultProps }) => ({ ...defaultDataTableProps, label, value, ...defaultProps }),
+  outlineInfo: component => component.label,
+  fields: {
+    components: { subsection: 'General', type: 'hidden' },
+    label: { subsection: 'General', label: 'Label', type: 'text' },
+    value: { subsection: 'General', label: 'Value', type: 'textBrowser', options: { onlyTypesOf: 'List<' } },
+    ...baseComponentFields
+  }
+};
+
+const UiBlock = ({ id, label, components, value }: UiComponentProps<DataTableProps>) => (
+  <div className='block-table'>
+    <div className='block-table__label'>
+      <span>{label}</span>
+      <span style={{ color: 'var(--N600)' }}>{value.length > 0 ? value : 'value is empty'}</span>
+    </div>
+    {components.length > 0 && (
+      <div className='block-table__columns'>
+        {components.map((column, index) => (
+          <ComponentBlock key={column.id} component={column} preId={components[index - 1]?.id} />
+        ))}
+      </div>
+    )}
+    {components.length === 0 && <DataTableColumnDialog id={id} initValue={value} />}
+  </div>
+);
+
+const DataTableColumnDialog = ({ id, initValue }: { id: string; initValue: string }) => {
```

Comment:
And now you can also rename this component to something like `EmptyDataTableColumn`